app/src/main/java/de/danoeh/antennapod/adapter/SelectableAdapter.java

```diff
@@ -0,0 +1,237 @@
+package de.danoeh.antennapod.adapter;
+
+import android.app.Activity;
+import android.util.AttributeSet;
+import android.util.SparseBooleanArray;
+import android.view.ActionMode;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import de.danoeh.antennapod.R;
+
+/**
+ * Used by Recyclerviews that need to provide ability to select items
+ * @param <T>
+ */
+class SelectableAdapter<T extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<T>{
+    private int selectedCount;
+
+    public SelectableAdapter(Activity activity) {
+        this.activity = activity;
+    }
+    @NonNull
+    @Override
+    public T onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        return null;
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull T holder, int position) {
+    }
+
+    @Override
+    public int getItemCount() {
+        return 0;
+    }
+
+    private ActionMode actionMode;
+    private SparseBooleanArray selectedItemPositions = new SparseBooleanArray();
```

Comment:
Doesn't this serve the same purpose as the `selectedIds` list in the child class? I think it would be better if keeping track of which (and how many) items are selected would be tracked in only one single location. Maybe using the RecyclerView method `getItemId()`?

I am a bit worried about using positions. What happens if the podcast gets refreshed and has a new episode while in multi-select mode? Then the positions shift around and I think it could break things. Storing IDs should me more stable than positions.